frontend/public/components/sidebars/resource-sidebar-samples.tsx

```diff
@@ -256,6 +269,85 @@ const ResourceSidebarSample: React.FC<ResourceSidebarSampleProps> = ({
   );
 };
 
+const lineHeight = 18;
+const PreviewYAML = ({ maxPreviewLines = 20, yaml }) => {
```

Comment:
pls move to a separate component file

Reply:
I followed the standard of this file. The `public` folder seems to be a bit of a wild west when it comes to standards... even the linter/prettier is turned off for it.

I could make more files for sure, but there are dozens of classes already in this file.

Reply:
I totally, agree. Then we can keep the components as-is and later think of a refactoring around the admin code.